lib/membrane_rtc_engine/endpoints/webrtc_endpoint.ex

```diff
@@ -774,206 +831,30 @@ defmodule Membrane.RTC.Engine.Endpoint.WebRTC do
-    {:ok, %{type: :renegotiate_tracks}}
-  end
-
-  defp deserialize(%{"type" => "prioritizeTrack"} = event) do
-    case event do
-      %{"type" => "prioritizeTrack", "data" => %{"trackId" => track_id}} ->
-        {:ok, %{type: :prioritize_track, data: %{track_id: track_id}}}
-    end
-  end
-
-  defp deserialize(%{"type" => "unprioritizeTrack"} = event) do
-    case event do
-      %{"type" => "unprioritizeTrack", "data" => %{"trackId" => track_id}} ->
-        {:ok, %{type: :unprioritize_track, data: %{track_id: track_id}}}
-    end
-  end
-
-  defp deserialize(%{"type" => "preferedVideoSizes"} = event) do
-    case event do
-      %{
-        "type" => "preferedVideoSizes",
-        "data" => %{
-          "bigScreens" => big_screens,
-          "mediumScreens" => medium_screens,
-          "smallScreens" => small_screens,
-          "allSameSize" => same_size?
-        }
-      } ->
-        {:ok,
-         %{
-           type: :prefered_video_sizes,
-           data: %{
-             big_screens: big_screens,
-             medium_screens: medium_screens,
-             small_screens: small_screens,
-             same_size?: same_size?
-           }
-         }}
-    end
-  end
-
-  defp deserialize(%{"type" => "candidate"} = event) do
-    case event do
-      %{
-        "type" => "candidate",
-        "data" => %{
-          "candidate" => candidate,
-          "sdpMLineIndex" => sdp_m_line_index
-        }
-      } ->
-        {:ok,
-         %{
-           type: :candidate,
-           data: %{
-             candidate: candidate,
-             sdp_m_line_index: sdp_m_line_index
-           }
-         }}
-
-      _other ->
-        {:error, :invalid_media_event}
-    end
-  end
-
-  defp deserialize(%{"type" => "sdpOffer"} = event) do
-    case event do
-      %{
-        "type" => "sdpOffer",
-        "data" => %{
-          "sdpOffer" => %{
-            "type" => "offer",
-            "sdp" => sdp
-          },
-          "trackIdToTrackMetadata" => track_id_to_track_metadata,
-          "midToTrackId" => mid_to_track_id
-        }
-      } ->
-        {:ok,
-         %{
-           type: :sdp_offer,
-           data: %{
-             sdp_offer: %{
-               type: :offer,
-               sdp: sdp
-             },
-             track_id_to_track_metadata: track_id_to_track_metadata,
-             mid_to_track_id: mid_to_track_id
-           }
-         }}
-
-      _other ->
-        {:error, :invalid_media_event}
-    end
-  end
-
-  defp deserialize(%{"type" => "setTargetTrackVariant"} = event) do
-    case event do
-      %{
-        "type" => "setTargetTrackVariant",
-        "data" => %{
-          "trackId" => tid,
-          "variant" => variant
-        }
-      } ->
-        {:ok, %{type: :set_target_track_variant, data: %{track_id: tid, variant: variant}}}
-
-      _other ->
-        {:error, :invalid_media_event}
+      case event do
+        %{type: :custom, data: data} -> {:ok, data}
+        e -> {:ok, e}
```

Comment:
Let's keep the convention and use `event` :smile:

You can also get rid of `with` and use single case